lib/mlir/Dialect/Clift/Transforms/ImportModelTypes.cpp

```diff
@@ -0,0 +1,55 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Diagnostics.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassRegistry.h"
+
+#include "revng-c/mlir/Dialect/Clift/IR/CliftOps.h"
+#include "revng-c/mlir/Dialect/Clift/Transforms/Passes.h"
+#include "revng-c/mlir/Dialect/Clift/Utils/ImportModel.h"
+
+namespace mlir {
+namespace clift {
+#define GEN_PASS_DEF_CLIFTIMPORTMODELTYPES
+#include "revng-c/mlir/Dialect/Clift/Transforms/Passes.h.inc"
+} // namespace clift
+} // namespace mlir
```

Comment:
And this is another piece of the magic. How does this part work?

Reply:
More tablegen generated stuff but for the implementation this time. Also unlike the header stuff, the implementation stuff only provides per pass opt-in macros. The header stuff also has these but also has macros to include all. (`GEN_PASS_DECL`, `GEN_PASS_REGISTRATION`). In particular on the implementation side it defines the `CliftImportModelTypesBase` base class from which the pass implementation is derived.